subiquity/server/controllers/tests/test_refresh.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2022 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from unittest import mock
+
+from subiquitycore.snapd import AsyncSnapd, get_fake_connection
+from subiquitycore.tests import SubiTestCase
+from subiquitycore.tests.mocks import make_app
+
+from subiquity.server.controllers import refresh as refresh_mod
+from subiquity.server.controllers.refresh import (
+    RefreshController,
+    SnapChannelSource,
+    )
+from subiquity.server import snapdapi
+
+
+class TestRefreshController(SubiTestCase):
+
+    def setUp(self):
+        self.app = make_app()
+        self.app.report_start_event = mock.Mock()
+        self.app.report_finish_event = mock.Mock()
+        self.app.note_data_for_apport = mock.Mock()
+        self.app.prober = mock.Mock()
+        self.app.snapdapi = snapdapi.make_api_client(
+            AsyncSnapd(get_fake_connection()))
+        self.rc = RefreshController(app=self.app)
+
+    async def test_configure_snapd_kernel_autoinstall(self):
+        # If a snap channel is supplied via kernel command line or
+        # autoinstall data, switch to it.
+
+        for src in SnapChannelSource.CMDLINE, SnapChannelSource.AUTOINSTALL:
+            with mock.patch.object(self.rc, 'get_refresh_channel') as grc:
+                grc.return_value = ('newchan', SnapChannelSource.CMDLINE)
```

Comment:
src is unused, intended for use here presumably

Reply:
oops, thanks